### PR TITLE
label TODOs

### PR DESCRIPTION
**Pull request by sternhenri**
**In this PR:**
- Handled a few rote TODOs, namely: 
  - `abortstatemsg` -> `abort`
  - `bytes` -> `[]byte`
  - stale TODOs
- labeled TODOs with likely DRIs

@zixuanzh see if you can knock any others out or pick other DRIs if you think there is better?

Thereafter good to merge.

### CODE REVIEW

File: actors/builtin/multisig/multisig_actor.go [hunk 1]
```diff
@@ -163,7 +162,7 @@ func (a *MultiSigActor) AddSigner(rt vmr.Runtime, params *AddSigner) *vmr.EmptyR
 	var st MultiSigActorState
 	rt.State().Transaction(&st, func() interface{} {
 		if st.isSigner(params.Signer) {
-			rt.AbortStateMsg("party is already a signer")
+			rt.Abort(exitcode.ErrIllegalState, "party is already a signer")
```
Comment (zixuanzh) [inline actors/builtin/multisig/multisig_actor.go]: ErrForbidden?

File: actors/builtin/multisig/multisig_actor.go [hunk 2]
```diff
@@ -217,11 +216,11 @@ func (a *MultiSigActor) SwapSigner(rt vmr.Runtime, params *SwapSignerParams) *vm
 	var st MultiSigActorState
 	rt.State().Transaction(&st, func() interface{} {
 		if !st.isSigner(params.From) {
-			rt.AbortStateMsg("Party not found")
+			rt.Abort(exitcode.ErrNotFound, "Party not found")
 		}
 
 		if !st.isSigner(params.To) {
-			rt.AbortStateMsg("Party already present")
+			rt.Abort(exitcode.ErrIllegalState, "Party already present")
```
Comment (zixuanzh) [inline actors/builtin/multisig/multisig_actor.go]: ErrForbidden?

File: actors/builtin/multisig/multisig_actor.go [hunk 3]
```diff
@@ -265,12 +264,12 @@ func (a *MultiSigActor) approveTransaction(rt vmr.Runtime, txnID TxnID) {
...
 		txn, err = st.getPendingTransaction(adt.AsStore(rt), txnID)
 		if err != nil {
-			rt.Abort(exitcode.ErrIllegalState, "failed to get transaction for approval: %v", err)
+			rt.Abort(exitcode.ErrNotFound, "failed to get transaction for approval: %v", err)
 		}
 		// abort duplicate approval
 		for _, previousApprover := range txn.Approved {
 			if previousApprover == rt.ImmediateCaller() {
-				rt.AbortStateMsg("already approved this message")
+				rt.Abort(exitcode.ErrIllegalState, "already approved this message")
```
Comment (zixuanzh) [inline actors/builtin/multisig/multisig_actor.go]: ErrForbidden?

File: actors/builtin/reward/reward_actor.go [hunk 4]
```diff
@@ -60,7 +61,7 @@ type RewardActorState struct {
 func (st *RewardActorState) _withdrawReward(rt vmr.Runtime, ownerAddr addr.Address) abi.TokenAmount {
 	rewards, found := st.RewardMap[ownerAddr]
 	if !found {
-		rt.AbortStateMsg("ra._withdrawReward: ownerAddr not found in RewardMap.")
+		rt.Abort(exitcode.ErrIllegalState, "ra._withdrawReward: ownerAddr not found in RewardMap.")
```
Comment (zixuanzh) [inline actors/builtin/reward/reward_actor.go]: ErrNotFound?

File: actors/builtin/reward/reward_actor.go [hunk 5]
```diff
@@ -123,13 +124,12 @@ func (a *RewardActor) AwardBlockReward(
...
 	pledgeReq := inds.PledgeCollateralReq(minerNominalPower)
 	currReward := inds.GetCurrBlockRewardForMiner(minerNominalPower, currPledge)
-	TODO() // BigInt
 
 	// 0 if over collateralized
 	underPledge := big.Max(big.Zero(), big.Sub(pledgeReq, currPledge))
 	rewardToGarnish := big.Min(currReward, underPledge)
 
-	TODO()
+	TODO() // ZX, JZ
```
Comment (zixuanzh) [inline actors/builtin/reward/reward_actor.go]: this has been handled

File: actors/builtin/storage_market/storage_market_actor.go [hunk 6]
```diff
@@ -339,31 +338,31 @@ func (a *StorageMarketActor) Constructor(rt Runtime) *vmr.EmptyReturn {
 
 func _rtAbortIfDealAlreadyProven(rt Runtime, deal OnChainDeal) {
 	if deal.SectorStartEpoch != epochUndefined {
-		rt.AbortStateMsg("Deal has already appeared in proven sector.")
+		rt.Abort(exitcode.ErrIllegalState, "Deal has already appeared in proven sector.")
```
Comment (zixuanzh) [inline actors/builtin/storage_market/storage_market_actor.go]: Not sure if these deal proposal related errors are IllegalState or IllegalArgument

File: actors/builtin/storage_market/storage_market_actor_state.go [hunk 7]
```diff
@@ -334,7 +334,7 @@ func (st *StorageMarketActorState) _rtGetOnChainDealOrAbort(rt Runtime, dealID a
 	var found bool
 	deal, dealP, found = st._getOnChainDeal(dealID)
 	if !found {
-		rt.AbortStateMsg("dealID not found in Deals.")
+		rt.Abort(exitcode.ErrIllegalState, "dealID not found in Deals.")
```
Comment (zixuanzh) [inline actors/builtin/storage_market/storage_market_actor_state.go]: ErrNotFound?

File: actors/builtin/storage_miner/storage_miner_actor.go [hunk 8]
```diff
@@ -281,12 +281,12 @@ func (a *StorageMinerActor) ExtendSectorExpiration(rt Runtime, sectorNumber abi.
...
 		sectorInfo, found := st.Sectors[sectorNumber]
 		if !found {
-			rt.AbortStateMsg("Sector not found")
+			rt.Abort(exitcode.ErrNotFound, "Sector not found")
 		}
 
 		extensionLength = newExpiration - sectorInfo.Info.Expiration
 		if extensionLength < 0 {
-			rt.AbortStateMsg("Cannot reduce sector expiration")
+			rt.Abort(exitcode.ErrForbidden, "Cannot reduce sector expiration")
```
Comment (zixuanzh) [inline actors/builtin/storage_miner/storage_miner_actor.go]: IllegalArgument? My understanding is that ErrForbidden refers to actors invoking actions that they do not have the right to

File: actors/builtin/methods.go [hunk 9]
```diff
@@ -8,7 +8,7 @@ const (
 	MethodSend        = abi.MethodNum(0)
 	MethodConstructor = abi.MethodNum(1)
 
-	// TODO: remove this once canonical method numbers are finalized
+	// TODO fin: remove this once canonical method numbers are finalized
```
Comment (zixuanzh) [inline actors/builtin/methods.go]: what's fin?

File: actors/builtin/shared.go [hunk 10]
```diff
@@ -30,7 +30,7 @@ const (
 // It is merely a method dispatch interface.
 type ActorCode interface {
 	//InvokeMethod(rt Runtime, method actor.MethodNum, params actor.MethodParams) InvocOutput
-	// TODO: method dispatch mechanism is deferred to implementations.
+	// TODO fin: method dispatch mechanism is deferred to implementations.
```
Comment (zixuanzh) [inline actors/builtin/shared.go]: ditto

File: actors/builtin/reward/reward_actor.go (hunk 4 above)
Comment (sternhenri) [reply actors/builtin/reward/reward_actor.go]: Yes!

File: actors/builtin/multisig/multisig_actor.go (hunk 1 above)
Comment (sternhenri) [reply actors/builtin/multisig/multisig_actor.go]: I took forbidden to mean a forbidden arg as in one that could not be valid (e.g. neg int or something) not one that happens not to be. I may be wrong. Which is it?

File: actors/builtin/storage_market/storage_market_actor.go (hunk 6 above)
Comment (sternhenri) [reply actors/builtin/storage_market/storage_market_actor.go]: mmh argument is legal would be my case, but doesnt work coz of state

File: actors/builtin/multisig/multisig_actor.go (hunk 1 above)
Comment (sternhenri) [reply actors/builtin/multisig/multisig_actor.go]: ah fair

Comment (sternhenri) [reply actors/builtin/multisig/multisig_actor.go]: but it's not you can't make him a signed, it's he already is...?

File: actors/builtin/reward/reward_actor.go (hunk 5 above)
Comment (sternhenri) [reply actors/builtin/reward/reward_actor.go]: Woohoo!

File: actors/builtin/storage_market/storage_market_actor.go (hunk 6 above)
Comment (sternhenri) [reply actors/builtin/storage_market/storage_market_actor.go]: Argument I think per re read

File: actors/builtin/storage_miner/storage_miner_actor.go (hunk 8 above)
Comment (sternhenri) [reply actors/builtin/storage_miner/storage_miner_actor.go]: ah ok. if that;s the case you're right (but a lot of your errForbidden suggestions above are wrong :b)

File: actors/builtin/methods.go (hunk 9 above)
Comment (sternhenri) [reply actors/builtin/methods.go]: sorry means can only be done when all else is closed ("end" in french)

Comment (anorth) [reply actors/builtin/methods.go]: I don't think it's helpful

File: actors/builtin/multisig/multisig_actor.go (hunk 2 above)
Comment (anorth) [reply actors/builtin/multisig/multisig_actor.go]: ErrIllegalArgument, and probably same for the one above.

File: actors/builtin/storage_miner/storage_miner_actor_state.go [hunk 11]
```diff
@@ -211,13 +211,13 @@ func MinerInfo_New(
 		PendingWorkerKey: WorkerKeyChange{},
 	}
 
-	TODO() // TODO: determine how to generate/validate VRF key and initialize other fields
+	TODO() // TODO anorth: determine how to generate/validate VRF key and initialize other fields
```
Comment (anorth) [inline actors/builtin/storage_miner/storage_miner_actor_state.go]: I don't know what this is talking about

File: actors/builtin/storage_power/storage_power_actor.go [hunk 12]
```diff
@@ -295,7 +297,7 @@ func (a *StoragePowerActor) OnMinerEnrollCronEvent(rt Runtime, eventEpoch abi.Ch
 }
 
 func (a *StoragePowerActor) ReportConsensusFault(rt Runtime, blockHeader1, blockHeader2 []byte, slashee addr.Address, faultEpoch abi.ChainEpoch, faultType ConsensusFaultType) *adt.EmptyValue {
-	// TODO: determine how to reward multiple reporters of the same fault within a single epoch.
+	// TODO: anorth determine how to reward multiple reporters of the same fault within a single epoch.
```
Comment (anorth) [inline actors/builtin/storage_power/storage_power_actor.go]: Nuh-uh! Cryptoecon problem.

File: actors/util/adt/map.go [hunk 13]
```diff
@@ -153,7 +154,7 @@ func (r rtStore) Context() context.Context {
 
 func (r rtStore) Get(ctx context.Context, c cid.Cid, out interface{}) error {
 	if !r.IpldGet(c, out.(vmr.CBORUnmarshaler)) {
-		r.AbortStateMsg("not found")
+		r.Abort(exitcode.ErrNotFound, "not found")
```
Comment (anorth) [inline actors/util/adt/map.go]: This should be illegal state, it means something is messed up in internal state.

File: actors/builtin/methods.go (hunk 9 above)
Comment (sternhenri) [reply actors/builtin/methods.go]: I don't understand that comment.

File: actors/builtin/storage_miner/storage_miner_actor_state.go (hunk 11 above)
Comment (sternhenri) [reply actors/builtin/storage_miner/storage_miner_actor_state.go]: Same interaction set about how you envision the worker key relating to the workerID which I've not understood yet

**Pull request merged**